File: dotnet/Cucumber.Messages.Specs/ProtocolVersionTest.cs
using Xunit;

namespace Cucumber.Messages.Specs;

public class ProtocolVersionTest
{
    [Fact]
    public void ProtocolVersionIsCorrect()
    {
        Assert.Matches(@"\d+\.\d+\.\d+(-[0-9A-Za-z-](\.[0-9A-Za-z-])*)?", Cucumber.Messages.ProtocolVersion.Version);
    }
}

File: java/src/test/java/io/cucumber/messages/ProtocolVersionTest.java
package io.cucumber.messages;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;


class ProtocolVersionTest {

    @Test
    void should_have_a_resource_bundle_version() {
        String version = ProtocolVersion.getVersion();
        assertThat(version).matches("\\d+\\.\\d+\\.\\d+(-SNAPSHOT)?");
    }

}
